### ISSUE TITLE
IAPHelper Not handling transactions at application launch? 

### ISSUE THREAD

**Issue by phasedarray** [saturngod/IAPHelper#34]
From the Apple's document In-App Purchase Best Practices
https://developer.apple.com/library/ios/technotes/tn2387/_index.html

Transaction observer SHOULD be added to SKPaymentQueue at application launch, the reason is that some transaction may be still in purchasing status when user closes the app.
But IAPHelper doesn't initialise it self at launch. Moreover buyProductCompleteBlock in IAPHelper isn't initialised until _buyProduct:onCompletion:_ or _restoreProductsWithCompletion_:  is invoked. So the updatedTransactions will be lost before any purchase or restore.

Am i correct?

**Comment by saturngod** [comment]
yes, it's correct. That why , I use `addTransactionObserver` at object allocate and `removeTransactionObserver` at dealloc of IAPHelper class.

**Comment by phasedarray** [comment]
but at that time, _buyProductCompleteBlock is still nil, customer code can't get any notification or callback.
